tools/toolchains/arm.py

```diff
@@ -63,7 +63,7 @@ def check_executable():
         return mbedToolchain.generic_check_executable("ARM", 'armcc', 2, 'bin')
 
     def __init__(self, target, notify=None, macros=None,
-                 build_profile=None, build_dir=None):
+                 build_profile=None, build_dir=None, coverage_patterns=None):
```

Comment:
The coverage stuff only applies to gcc-arm so this class shouldn't need to change.

Reply:
Sorry for the late reply, now back from the holidays and continue on this PR.
The reason for the change in the ARM and IAR is because all of the toolchains shared the same API. e.g. if uses:
```
mbed test -t ARM --coverage-filter "somefile"
```

the build tool will break and throw stack trace, saying  `coverage_patterns` option is not defined.
But with this change of adding the option in both `ARM` and `IAR`. the tools will not break, but ignore this option and throw a warning, saying the "--coverage-filter" option is not supported.

This also allows the tools tests to pass.